### PR TITLE
Support CSV/TSV/CustomSeparated raw streaming

### PR DESCRIPTION
**Pull request by slvrtrn**
Partially implements #28 

Adds 
```
  'CSV',
  'CSVWithNames',
  'CSVWithNamesAndTypes',
  'TabSeparated',
  'TabSeparatedRaw',
  'TabSeparatedWithNames',
  'TabSeparatedWithNamesAndTypes',
  'CustomSeparated',
  'CustomSeparatedWithNames',
  'CustomSeparatedWithNamesAndTypes',
```
as possible raw streaming formats

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -199,14 +199,19 @@ function formatCommandQuery(query: string, format: DataFormat | false): string {
...
 ): void {
   if (Array.isArray(values) === false && isStream(values) === false) {
     throw new Error(
       'Insert expected "values" to be an array or a stream of values.'
     )
   }
 
+  if (format.startsWith('CSV')) {
+    return
+  }
```
Comment (slvrtrn) [inline src/client.ts]: so it works just fine with these two short-circuits (this one and in the `formatter.ts`) as it seems. We can leave it as is or think about a better insert values abstraction as discussed.

Comment (mshustov) [reply src/client.ts]: The current approach doesn't allow inserting `JSON*` formats as strings/bytes. Let's imagine a user reads a `ndjson` file to insert into DB. Do they need to parse it to convert it into JSON now?

File: __tests__/integration/insert_csv_file_stream.test.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    await client.insert({
+      table: tableName,
+      values: stream,
+      format: 'CSVWithNamesAndTypes',
+    })
+    await assertInsertedValues()
+  })
+
+  it('should throw in case of invalid CSV format', async () => {
+    const stream = createFileStream('insert_csv_invalid_format.csv')
```
Comment (mshustov) [inline __tests__/integration/insert_csv_file_stream.test.ts]: Do we need to create files? A Readable stream can be instantiated with a list of strings
```js
Stream.from(['foobar,42,,'])
```

File: __tests__/integration/insert_stream_raw_formats.test.ts [hunk 3]
```diff
@@ -0,0 +1,241 @@
+import { createTestClient, guid } from '../utils'
+import type { ClickHouseClient, ClickHouseSettings } from '../../src'
+import { createSimpleTable } from './fixtures/simple_table'
+import Stream from 'stream'
+
+// FIXME `npm run test -- --detectOpenHandles`
+//  Jest has detected the following 13 open handles
+//  potentially keeping Jest from exiting
```
Comment (slvrtrn) [inline __tests__/integration/insert_stream_raw_formats.test.ts:8]: Still a mystery why this is happening

File: src/data_formatter/formatter.ts [hunk 4]
```diff
@@ -80,6 +86,12 @@ export function decode(text: string, format: DataFormat): any {
  * @returns string
  */
 export function encode(value: any, format: DataFormat): string {
+  // TODO: add
+  //   'JSONStringsEachRow',
+  //   'JSONCompactEachRowWithNames',
+  //   'JSONCompactEachRowWithNamesAndTypes',
+  //   'JSONCompactStringsEachRowWithNames',
+  //   'JSONCompactStringsEachRowWithNamesAndTypes',
```
Comment (slvrtrn) [inline src/data_formatter/formatter.ts:94]: I guess support for these formats should also go into this `encode` function as a follow-up, WDYT @mshustov?

**Pull request merged**